backend/src/main/scala/sbt/internal/inc/bloop/internal/BloopHighLevelCompiler.scala

```diff
@@ -217,19 +217,29 @@ final class BloopHighLevelCompiler(
       /* `Mixed` order defaults to `ScalaThenJava` behaviour.
        * See https://github.com/sbt/zinc/issues/234. */
       if (setup.order == CompileOrder.JavaThenScala) {
-        fireJavaCompilation.flatMap {
-          case JavaSignal.ContinueCompilation => compileJava.flatMap(_ => compileScala)
-          case JavaSignal.FailFastCompilation(failedProjects) =>
-            throw new StopPipelining(failedProjects)
-        }
-      } else {
-        compileScala.flatMap { _ =>
+        if (javaSources.isEmpty) {
```

Comment:
I think you could avoid code duplication if this `if` was the outer one, as they both seem to be the same (except that `()` argument which I believe is a mistake). WDYT?

Reply:
Yeah, you're right :wink: